### PR TITLE
View vars etc

### PR DESCRIPTION
**Pull request by davidlehn**
- Removing more views stuff.
- Changing `optimize` to `bundle`.  Similar to older `optimize` command and `minify` option.  But now uses `bundle` and need to set `views.bundle.mode` to `"production"` or use `--bundle-mode` option.
- I don't know if the basic default error templates are useful or not. Ideally no one would see them, but that may not be reality.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,9 +22,9 @@
   - If webpack has trouble with `crypto`, check if its usage can be easily
     avoided. If not, one technique is to add `bedrock-webpack` config that
     aliases `crypto` to a file that just throws an error. Some libraries use a
-    try/catch on `crypto` to choose Node.js or browser paths. Try to move away
-    from this if possible.
-  - In a top level file, like `components/app.js` add as needed:
+    try/catch on `require('crypto')` to choose Node.js or browser paths. Try to
+    move away from this if possible.
+  - In a top level file exists, like `components/app.js`, add as needed:
```
Comment (dlongley) [inline CHANGELOG.md]: Did you mean this?

```suggestion
  - If a top level file exists, like `components/app.js`, add as needed:
```

File: views/main.html [hunk 2]
```diff
@@ -1 +1,25 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <base href="/" />
+    <title></title>
+    <link rel="shortcut icon" href="/favicon.ico"
+      media="none" onload="if(media!='all')media='all'">
+    <script defer src="/static/js/bundle.js"></script>
+  </head>
+
+  <body class="br-body">
+    <br-app>
```
Comment (dlongley) [inline views/main.html]:
```suggestion
    <br-app></br-app>
```

File: lib/config.js [hunk 3]
```diff
@@ -42,110 +37,23 @@ cc('views.bundle.paths.input.root', () => path.join(
...
-
-// branding config
-config.views.brand = {};
-config.views.brand.name = '';
-
-// TODO: backwards compatibility, remove brand at top level of config
-config.brand = config.views.brand;
-
 config.views.cache = false;
+// and consolidate.js engine can be used as long as proper lib is installed.
```
Comment (mattcollier) [inline lib/config.js]: Are these two lines related?  How is lodash a views engine?

File: lib/views.js [hunk 4]
```diff
@@ -89,60 +25,36 @@ function configure(app) {
...
-        loader: swigLoaders.multipath({base: paths})
-      });
-      view = swig.compileFile(path)(options);
-    } catch(err) {
-      return callback(err);
-    }
-    callback(null, view);
-  });
+  // add the default template engine
+  // TODO: use configuration to allow other engines
```
Comment (mattcollier) [inline lib/views.js]: Looks like config is being used now, is that right?

File: lib/views.js [hunk 5]
```diff
@@ -89,60 +25,36 @@ function configure(app) {
...
+  app.engine('html', consolidate[bedrock.config.views.engine]);
   app.set('view engine', 'html');
   app.set('views', paths);
 
   if(bedrock.config.views.serviceUnavailable) {
     // service unavailable for all requests
-    app.all('*', function(req, res, next) {
+    app.all('*', function(req, res/*, next*/) {
```
Comment (mattcollier) [inline lib/views.js]: arrow function?

**Review by mattcollier — COMMENTED** [review]
just some questions/comments.

File: lib/config.js (hunk 3 above)
Comment (davidlehn) [reply lib/config.js]: Typo there: s/and/any/.  consolidate.js wraps lots of template engines.  lodash templates are one choice.  Not as featureful as swig or similar but seemed a ok choice if we just pass in a simple var.

File: lib/views.js (hunk 4 above)
Comment (davidlehn) [reply lib/views.js]: Yup. Forgot to remove comment.

File: lib/views.js (hunk 5 above)
Comment (davidlehn) [reply lib/views.js]: I didn't arrowize all the things.  Should I?
